### PR TITLE
DEM-1260 Update submissions service & file-detail for messaging

### PR DESCRIPTION
**Pull request by samsep**
pign @aselbie @vietnogi

### CODE REVIEW

File: src/scripts/file-detail.controller.coffee [hunk 1]
```diff
@@ -37,8 +37,13 @@ FileDetailController = (helpers, $scope, $rootScope, SubmissionsService, UserV3S
     $scope.$on '$destroy', ->
       destroySubmissionsListener()
 
+    $scope.$watch UserV3Service.getCurrentUser, (user) ->
+      if user
+        vm.userId = user.id
```
Comment (vietnogi) [inline src/scripts/file-detail.controller.coffee]: `vm.userId = user?.id` ?

File: src/scripts/submissions.service.coffee [hunk 2]
```diff
@@ -36,10 +36,10 @@ srv = ($rootScope, helpers, StepsAPIService, SubmissionsAPIService, MessagesAPIS
       apiCall: apiCall
 
   markMessagesAsRead = (submissionId, fileId, userId) ->
-    submission     = submissions.findWhere { id: submissionId }
+    submission     = submissions.findWhere({ id: submissionId })[0]
```
Comment (vietnogi) [inline src/scripts/submissions.service.coffee]: `submissions.findWhere(id: submissionId)[0]`

File: src/scripts/submissions.service.coffee [hunk 3]
```diff
@@ -63,8 +63,9 @@ srv = ($rootScope, helpers, StepsAPIService, SubmissionsAPIService, MessagesAPIS
           read: true
 
   sendMessage = (submissionId, fileId, message, userId) ->
-    currentSubmissions = helpers.findInCollection submissions, 'id', submissionId
-    currentFile        = helpers.findInCollection currentSubmissions.files, 'id', fileId
+    currentSubmissions = submissions.get()
+    currentSubmission = helpers.findInCollection currentSubmissions, 'id', submissionId
```
Comment (vietnogi) [inline src/scripts/submissions.service.coffee]: the heck are `helpers`? can we have a less generic name?

Comment (aselbie) [inline src/scripts/submissions.service.coffee]: They are inject here as ``helpers``, but are actually defined as ``SubmissionsHelpers``. It's a bunch of functions that I found myself using a lot through submissions so I abstracted them out into a shared file.

Comment (vietnogi) [inline src/scripts/submissions.service.coffee]: i see, 
`
$.grep currentSubmissions, (submission) -> 
  submission.id == submissionId
`

**Pull request merged**